src/vs/workbench/services/policies/common/multiplexPolicyService.ts

```diff
@@ -0,0 +1,51 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IStringDictionary } from '../../../../base/common/collections.js';
+import { PolicyName } from '../../../../base/common/policy.js';
+import { ILogService } from '../../../../platform/log/common/log.js';
+import { AbstractPolicyService, IPolicyService, PolicyDefinition } from '../../../../platform/policy/common/policy.js';
+
+export class MultiplexPolicyService extends AbstractPolicyService implements IPolicyService {
+
+	constructor(
+		private readonly policyServices: ReadonlyArray<IPolicyService>,
+		@ILogService private readonly logService: ILogService,
+	) {
+		super();
+
+		// Forward policy changes from child services
+		for (const service of policyServices) {
+			this._register(service.onDidChange(names => this.onDidChangePolicies(names, service)));
+		}
+	}
+
+	protected async _updatePolicyDefinitions(policyDefinitions: IStringDictionary<PolicyDefinition>): Promise<void> {
```

Comment:
I am not sure if this service is implemented properly. Where are we initialising this service with initial policy values?

Reply:
Wouldn't it be in https://github.com/microsoft/vscode/blob/5f0269a4c9bc731eb9161340d8a1377fb130e88e/src/vs/platform/policy/common/policy.ts#L41-L42, via https://github.com/microsoft/vscode/blob/5f0269a4c9bc731eb9161340d8a1377fb130e88e/src/vs/platform/configuration/common/configurations.ts#L149-L150

So for extensions it would get configured here once the extension loads in:
https://github.com/microsoft/vscode/blob/5f0269a4c9bc731eb9161340d8a1377fb130e88e/src/vs/platform/configuration/common/configurations.ts#L115-L116

Reply:
I mean, every policy service has initial state [here](https://github.com/microsoft/vscode/blob/38e2bc4e6813e9f7ec0ae9b8b4ea5d9fa71d2bc9/src/vs/platform/policy/common/policy.ts#L30-L31). This means, native policy service has initial policy definitions and policy values, similarly for default account policy service. So, multiplex policy service should contain the initial state. For example, if I ask for policy value immediately after creating it, I should get the value defined in the native policy service or account policy service. I see this is missing right now.